repo/fsrepo.go

```diff
@@ -458,19 +467,35 @@ func fileExists(file string) bool {
 }
 
 // StagingDir satisfies node.SectorDirs
-func (r *FSRepo) StagingDir() string {
-	return path.Join(r.path, "staging")
+func (r *FSRepo) StagingDir() (string, error) {
+	// Default is to keep sector data alongside node repo data.
+	if r.cfg.SectorBase.RootDir == "" {
+		return path.Join(r.homePath, "sectors/staging"), nil
+	}
+	// If not using default check for writeability
+	if err := checkWritable(r.cfg.SectorBase.RootDir); err != nil {
+		return "", nil
+	}
+	return path.Join(r.cfg.SectorBase.RootDir, "staging"), nil
 }
 
 // SealedDir satisfies node.SectorDirs
-func (r *FSRepo) SealedDir() string {
-	return path.Join(r.path, "sealed")
+func (r *FSRepo) SealedDir() (string, error) {
+	// Default is to keep sector data alongside node repo data.
+	if r.cfg.SectorBase.RootDir == "" {
+		return path.Join(r.homePath, "sectors/sealed"), nil
+	}
+	// If not using default check for writeability
+	if err := checkWritable(r.cfg.SectorBase.RootDir); err != nil {
+		return "", nil
+	}
+	return path.Join(r.cfg.SectorBase.RootDir, "sealed"), nil
```

Comment:
As noted elsewhere, remove this responsibility from "fsrepo". I would suggest the permission checks are out of place here, too.